ee/tables/windowsupdatetable/windowsupdate_cached.go

```diff
@@ -0,0 +1,122 @@
+	"github.com/kolide/launcher/ee/tables/dataflattentable"
+	"github.com/kolide/launcher/ee/tables/tablehelpers"
+	"github.com/kolide/launcher/ee/tables/tablewrapper"
+	"github.com/osquery/osquery-go/plugin/table"
+)
+
+const (
+	cachedDataTTL = 6 * time.Hour
+)
+
+type CachedWindowsUpdatesTable struct {
+	slogger    *slog.Logger
+	cacheStore types.Getter
+	name       string
+}
+
+func CachedWindowsUpdatesTablePlugin(flags types.Flags, slogger *slog.Logger, cacheStore types.Getter) *table.Plugin {
+	columns := dataflattentable.Columns(
+		table.TextColumn("locale"),
+		table.IntegerColumn("is_default"),
+		table.IntegerColumn("is_cached"),
+		table.IntegerColumn("age"),
+	)
+
+	t := &CachedWindowsUpdatesTable{
+		slogger:    slogger.With("name", "kolide_windows_updates_cached"),
+		cacheStore: cacheStore,
+		name:       "kolide_windows_updates_cached",
+	}
+
+	return tablewrapper.New(flags, slogger, t.name, columns, t.generateFromCachedData)
+}
+
+func (c *CachedWindowsUpdatesTable) generateFromCachedData(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+	ctx, span := observability.StartSpan(ctx, "table_name", c.name)
+	defer span.End()
+
+	var results []map[string]string
+
+	for _, locale := range tablehelpers.GetConstraints(queryContext, "locale", tablehelpers.WithDefaults(defaultLocale)) {
+		rawLocaleResults, err := c.cacheStore.Get([]byte(locale))
+		if err != nil {
+			c.slogger.Log(ctx, slog.LevelWarn,
+				"could not get cached data for locale",
+				"locale", locale,
+				"err", err,
+			)
+			continue
+		}
+		if len(rawLocaleResults) == 0 {
+			c.slogger.Log(ctx, slog.LevelWarn,
+				"no cached data set for locale",
+				"locale", locale,
+			)
+			continue
+		}
+
+		var res cachedQueryResults
+		if err := json.Unmarshal(rawLocaleResults, &res); err != nil {
+			c.slogger.Log(ctx, slog.LevelWarn,
+				"could not unmarshal cached data for locale",
+				"locale", locale,
+				"err", err,
+			)
+			continue
+		}
+
+		if res.QueryTime.Add(cachedDataTTL).Before(time.Now()) {
+			c.slogger.Log(ctx, slog.LevelWarn,
+				"cached data is expired, not using",
+				"locale", locale,
+				"cached_data_query_time", res.QueryTime.String(),
+			)
+			continue
+		}
+
+		resultsAgeInSecondsStr := strconv.Itoa(int(time.Now().Unix() - res.QueryTime.Unix()))
+
+		for _, dataQuery := range tablehelpers.GetConstraints(queryContext, "query", tablehelpers.WithDefaults("*")) {
+			flattenOpts := []dataflatten.FlattenOpts{
+				dataflatten.WithSlogger(c.slogger),
+				dataflatten.WithQuery(strings.Split(dataQuery, "/")),
+			}
+
+			flatData, err := dataflatten.Json(res.Results.RawResults, flattenOpts...)
+			if err != nil {
+				c.slogger.Log(ctx, slog.LevelInfo,
+					"flatten failed",
+					"err", err,
+				)
+				continue
+			}
+
+			rowData := map[string]string{
+				"locale":     res.Results.Locale,
+				"is_default": strconv.Itoa(res.Results.IsDefaultLocale),
+				"is_cached":  "1",
```

Comment:
if I'm reading this correctly, this will always be "1"? We need this and `age`? I can live with it =)

Reply:
ehh nvm .... I see @directionless suggested it, I assume there is a downstream sql thingy that this helps with

Reply:
seph asked for em here! https://github.com/kolide/launcher/pull/2230#discussion_r2064509118

We don't need them at the moment, but implementation is still a bit up in the air so we might want them later

Reply:
Ah sorry, I only saw your first comment when I responded!

Reply:
`1` is true. 

I don't have an exact need, but it feels _correct_ to set it, even though it's a bit superfluous with `age`. :shrug: